src/main/java/software/amazon/encryption/s3/S3EncryptionClient.java

```diff
@@ -407,7 +440,13 @@ public UploadPartResponse uploadPart(UploadPartRequest request, RequestBody requ
     @Override
     public CompleteMultipartUploadResponse completeMultipartUpload(CompleteMultipartUploadRequest request)
             throws AwsServiceException, SdkClientException {
-        return _multipartPipeline.completeMultipartUpload(request);
+        try {
+            return _multipartPipeline.completeMultipartUpload(request);
+        } catch (CompletionException e) {
+            throw new S3EncryptionClientException(e.getCause().getMessage(), e.getCause());
+        } catch (Exception e) {
+            throw new S3EncryptionClientException("Unable to delete objects.", e);
```

Comment:
Same here, the error message here is wrong for this method `completeMultipartUpload`. Instead, we should be throwing `e.getMessage()`.